lib/widgets/matrix.dart

```diff
@@ -351,13 +353,13 @@ class MatrixState extends State<Matrix> with WidgetsBindingObserver {
           ),
         );
 
-        if (state != LoginState.loggedIn) {
-          navigatorContext?.go('/rooms');
-        }
+        navigatorContext?.go('/rooms');
       } else {
         setUpToMServicesInLogin(c);
         if (state == LoginState.loggedIn) {
           navigatorContext?.go('/rooms');
+        } else {
+          await getIt.get<HiveCollectionToMDatabase>().clear();
```

Comment:
Why do it here? Why not do it when logout?

Reply:
because "onLoginStateChanged" catches everytime the client logout or login, IMO it's better flow and cleaner

Reply:
1. How `HiveCollectionsDatabase` work?

Reply:
`HiveCollectionsDatabase (FlutterHiveCollectionsDatabase)` is initialized on the Matrix `createClient` function and destroyed in the logout function (`client.dart - 562 - clear()`)

We can call Client.clear

Reply:
https://github.com/linagora/twake-on-matrix/pull/543/commits/d5ba17909f7b93a60f998960eba4abbbb7315ac9